### PR TITLE
Support Small index config

### LINKED ISSUE #16730: [Bug]: Support segcore small index config
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

small index is created in chunk, for the acceleration of growing segment search.
However, we didn't carefully design small index for all parameters, and the nlist, nprobe is hard coded.
In this pr we are supporting config nlist and nprobe, later we will support more configs in Segcore and Knowhere.

### Expected Behavior

nlist, nprobe should be configured

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by xiaofan-luan**
fix #16730  
Signed-off-by: xiaofan-luan <xiaofan.luan@zilliz.com>

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -176,10 +176,12 @@ queryNode:
...
     searchResult:
       recvBufSize: 64 # msgPack channel buffer size
-  # Segcore will divide a segment into multiple chunks.
+  # Segcore will divide a segment into multiple chunks to enbale small index
   segcore:
     chunkRows: 32768 # The number of vectors in a chunk.
-
+    smallIndex:
+      nlist: 256 # small index nlist, recommend to set sqrt(chunkRows), must smaller than chunkRows/8
+      nprobe: 16 # nprobe to search small index, based on your accuracy requirement, must smaller than nlist
```
Comment (cydrain) [inline configs/milvus.yaml:184]: in general, we need not limit "nprobe <= nlist", we can handle it in source code like this:
nprobe = std::min(nprobe, nlist)
Just like how faiss handle index parameters for IVF

File: internal/util/paramtable/component_param.go [hunk 2]
```diff
@@ -797,8 +800,34 @@ func (p *queryNodeConfig) initGracefulTime() {
...
+func (p *queryNodeConfig) initSmallIndexParams() {
 	p.ChunkRows = p.Base.ParseInt64WithDefault("queryNode.segcore.chunkRows", 32768)
+	if p.ChunkRows < 1024 {
+		log.Warn("chunk rows can not be less than 1024, force set to 1024", zap.Any("current", p.ChunkRows))
+		p.ChunkRows = 1024
+	}
+
+	// default NList is the first nlist
+	var defaultNList int64
+	for i := int64(0); i < p.ChunkRows; i++ {
```
Comment (cydrain) [inline internal/util/paramtable/component_param.go:812]: i should start from 10

File: configs/milvus.yaml (hunk 1 above)
Comment (xiaofan-luan) [reply configs/milvus.yaml:184]: it is already handled by go side

	defaultNprobe := p.SmallIndexNlist / 16
	p.SmallIndexNProbe = p.Base.ParseInt64WithDefault("queryNode.segcore.smallIndex.nprobe", defaultNprobe)
	if p.SmallIndexNProbe > p.SmallIndexNlist {
		log.Warn("small index nprobe must smaller than nlist, force set to", zap.Any("nprobe", p.SmallIndexNlist))
		p.SmallIndexNProbe = p.SmallIndexNlist
	}

**Pull request merged**